cmake/modules/CUDA.cmake

```diff
@@ -64,6 +64,7 @@ if(USE_CUDA)
     message(STATUS "Build with Thrust support")
     cmake_minimum_required(VERSION 3.13) # to compile CUDA code
     enable_language(CUDA)
+    set(CMAKE_CUDA_ARCHITECTURES "80;75")
```

Comment:
Why do we want to explicitly set the cuda arch here?

Reply:
The is due to an odd behavior of cmake that I don't understand. If I remove this and set `USE_THRUST`, I get
```
CMake Error in CMakeLists.txt:
  CUDA_ARCHITECTURES is empty for target "tvm_runtime_objs".
```
. Previously I didn't encounter this problem when I was using `USE_THRUST` (a couple of years ago).

This doesn't apply to `vllm.cmake` for some reason. But there, if I don't have `set(CMAKE_CUDA_ARCHITECTURES "80;75")`, I get 
```
error   : Feature 'f16 arithemetic and compare instructions' requires .target sm_53 or higher
```
during build.

Reply:
@masahi 

CMAKE_CUDA_ARCHITECTURES is designed to be eprovided by user at cmake config time.
* There are also values like 'auto', 'all' see [CUDA_ARCHITECTURES](https://cmake.org/cmake/help/latest/prop_tgt/CUDA_ARCHITECTURES.html)
* Better throw a warning like: 'Using 'all' for CUDAARCHS, use CMAKE_CUDA_ARCHITECTURES to override'.
* Also it is initialized by env var [CUDAARCHS](https://cmake.org/cmake/help/latest/variable/CMAKE_CUDA_ARCHITECTURES.html) and picked up from there.